hermes/views.py

```diff
@@ -178,40 +183,41 @@ def submit_to_hop(request, message):
         hop_auth: Auth = hopskotch.get_hermes_hop_authorization()
 
     logger.info(f'submit_to_hop User {request.user} with credentials {hop_auth.username}')
-
     logger.debug(f'submit_to_hop request: {request}')
     logger.debug(f'submit_to_hop request.data: {request.data}')
-    logger.debug(f'submit_to_hop s.write => message: {message}')
 
     try:
         topic = request.data['topic']
+        # Add the _sender in the header with our auth here since we originally generate it without an auth
+        headers.append(("_sender", hop_auth.username.encode("utf-8")))
         stream = Stream(auth=hop_auth)
         # open for write ('w') returns a hop.io.Producer instance
         with stream.open(f'kafka://kafka.scimma.org/{topic}', 'w') as producer:
-            metadata = {'topic': topic}
-            payload, headers = producer.pack(message, metadata)
-            message_uuid_tuple = next((item for item in headers if item[0] == '_id'), None)
-            message_uuid = None
-            if message_uuid_tuple:
-                message_uuid = uuid.UUID(bytes=message_uuid_tuple[1])
             producer.write_raw(payload, headers)
```

Comment:
Do we not need the UUID still in this case? Don't we need a uuid to reference our version of the message? Or are we only listening to their stream

Reply:
the uuid is placed in the headers that are passed into this method when they are generated